app/Models/Application.php

```
@@ -2027,35 +2026,970 @@ public function generateConfig($is_json = false)
-    public function setConfig($config)
+    /**
+     * Apply configuration from a coolify.json config array or JSON string.
+     *
+     * @param  array|string  $config  The configuration to apply
+     * @param  bool  $fromRepository  If true, the config comes from a repository's coolify.json file.
+     *                                In this case, the 'source' section (repository, branch, commit_sha)
+     *                                is ignored since these are already set from the actual git source.
+     *                                If false (default), the config is from a copy-paste import and
+     *                                source settings will be applied if present.
+     */
+    public function setConfig(array|string $config, bool $fromRepository = false): void
```

PHPMD (2.15.0)

The method setConfig has a boolean flag argument $fromRepository, which is a certain sign of a Single Responsibility Principle violation. (undefined)

(BooleanArgumentFlag)